server/routes/Router.js

```diff
@@ -0,0 +1,62 @@
+const express = require('express')
+const router = express.Router()
+const City = require('../models/City')
+
+const { refactorCityObj, weatherApiRQ } = require('../helpers/functions')
+
+router.get('/city/:cityName', async (req, res) => {
+    const { cityName } = req.params
+    const { lat, long } = req.query
+    const data = await weatherApiRQ(cityName, lat, long)
+    res.send(refactorCityObj(data, 'new'))
+})
+
+router.get('/cities', (req, res) => {
+    City
+        .find({})
+        .exec((e, d) => { e ? res.send(e) : res.send(d) })
+})
+
+router.post('/city', (req, res) => {
+    const { name, temperature, condition, conditionPic } = req.body
+    const c = new City({
+        name,
+        temperature,
+        condition,
+        conditionPic,
+        saved: true
+    })
+    c
+        .save()
+        .then((d) => { res.send({ city: d, msg: 'Saved' }) })
+        .catch((e) => { res.send(e) })
+})
+
+router.delete('/city/:cityName', (req, res) => {
+    const { cityName } = req.params
+    City
+        .findOneAndDelete({ name: cityName })
+        .exec((e, d) => { e ? res.send(e) : res.send({ city: d, msg: 'Deleted' }) })
+})
```

Comment:
you used the data-id with the _id from MongoDB but you don't use it to delete a city...